src/client/input/header/gyrospace.h

```diff
@@ -0,0 +1,393 @@
+        return;
+    
+    if (isnan(accel.x) || isnan(accel.y) || isnan(accel.z) ||
+        isnan(gyroRotation.x) || isnan(gyroRotation.y) || isnan(gyroRotation.z))
+        return;
+    
+    Vector3 accelNorm = Vec3_Normalize(accel);
+    
+    if (Vec3_IsZero(accelNorm)) {
+        Vector3 rotationDelta = Vec3_Cross(gyroRotation, gravNorm);
+        gravNorm = Vec3_Add(gravNorm, Vec3_Scale(rotationDelta, deltaTime));
+    } else {
+        Vector3 rotationDelta = Vec3_Cross(gyroRotation, gravNorm);
+        Vector3 rotatedGravity = Vec3_Add(gravNorm, Vec3_Scale(rotationDelta, deltaTime));
+        gravNorm = Vec3_Lerp(rotatedGravity, accelNorm, fusionFactor);
+    }
+    
+    if (!Vec3_IsZero(gravNorm))
+        gravNorm = Vec3_Normalize(gravNorm);
+    else
+        gravNorm = Vec3_New(0.0f, 1.0f, 0.0f);
+}
+
+/** Sets the global gravity vector manually. */
+static inline void SetGravityVector(float x, float y, float z) {
+    if (isnan(x) || isnan(y) || isnan(z))
+        return;
+    
+    Vector3 newGrav = Vec3_New(x, y, z);
+    if (Vec3_Magnitude(newGrav) < EPSILON)
+        return;
+    
+    gravNorm = Vec3_Normalize(newGrav);
+}
+
+/** Returns the current global gravity vector. */
+static inline Vector3 GetGravityVector(void) {
+    return gravNorm;
+}
+ 
+ // Dynamic Orientation Adjustment
+ 
+ /**
+  * Adapts gyro input for switching between normal controller grip and handheld-style grip.
+  * This function takes yaw, pitch, and roll inputs and adjusts them to maintain consistent orientation.
+  */
+Vector3 TransformWithDynamicOrientation(float yaw_input, float pitch_input, float roll_input) {
+    // Validate inputs.
+    if (isnan(yaw_input) || isnan(pitch_input) || isnan(roll_input)) {
+        return Vec3_New(0.0f, 0.0f, 0.0f);
+    }
+
+    // Clamp inputs to prevent extreme values.
+    yaw_input   = clamp(yaw_input, -360.0f, 360.0f);
+    pitch_input = clamp(pitch_input, -360.0f, 360.0f);
+    roll_input  = clamp(roll_input, -360.0f, 360.0f);
+
+    // Combine inputs into a single vector.
+    Vector3 combined = Vec3_New(yaw_input, pitch_input, roll_input);
+    
+    // Remove the gravity component to adjust orientation.
+    Vector3 adjusted = Vec3_Subtract(combined, Vec3_Scale(gravNorm, Vec3_Dot(combined, gravNorm)));
+    
+    // Normalize if necessary.
+    if (Vec3_Magnitude(adjusted) > EPSILON)
+        adjusted = Vec3_Normalize(adjusted);
+    
+    // Dynamic roll compensation based on gravity.
+    float adjRoll = roll_input - Vec3_Dot(gravNorm, Vec3_New(0.0f, 0.0f, roll_input));
+    
+    return Vec3_New(adjusted.x, adjusted.y, adjRoll);
+}
+ 
+ // Gyro Space Transformation Function
+ 
+ /**
+  * These functions transform gyro inputs into different spaces (Local, Player, World)
+  * while handling sensitivity adjustments and gravity alignment.
+  */
+ 
+ 
+// C wrapper for gyro transformation functions, ensuring compatibility with both C and C++.
+#ifdef __cplusplus
+extern "C" {
```

Comment:
redundant, you already have `extern "C" {` at the beginning of the file, affecting the whole file (because the corresponding `}` is at the end of the file)